erpnext/stock/report/batch_wise_balance_history/batch_wise_balance_history.js

```diff
@@ -18,6 +26,47 @@ frappe.query_reports["Batch-Wise Balance History"] = {
 			"width": "80",
 			"default": frappe.datetime.get_today(),
 			"reqd": 1
-		}
+		},
+		{
+			"fieldname":"item_code",
+			"label": __("Item Code"),
+			"fieldtype": "Link",
+			"options": "Item",
+			"get_query": function() {
+				return {
+					filters: {
+						"has_batch_no": 1
+					}
+				}
+			}
+		},
+		{
+			"fieldname":"warehouse",
+			"label": __("Warehouse"),
+			"fieldtype": "Link",
+			"options": "Warehouse",
+			"get_query": function() {
+				let company = frappe.query_report.get_filter_value('company');
+				return {
+					filters: {
+						"company": company
+					}
+				}
+			}
+		},
+		{
+			"fieldname":"batch_no",
+			"label": __("Batch No"),
+			"fieldtype": "Link",
+			"options": "Batch",
+			"get_query": function() {
+				let item_code = frappe.query_report.get_filter_value('item_code');
+				return {
+					filters: {
+						"item": item_code
+					}
+				}
+			}
+		},
```

Comment:
If Item filter isn't populated, the batch dropdown is empty. Either ask user to select an item if empty, on batch filter get_query. Or make batch filter visible only if item is populated.